File: ProyectoMysql/server/BusinessLayer/Entidad.py

from DataLayer.EntidadDB import *
from EntityLayer.EmpresaEntity import *
from Utilidades.Conexion.configMysql import StartTransaction, EndTransaction, Restore


class Entidad:

    def PersonaNaturalObtenerMain():
        try:
            return EntidadDB.PersonaNaturalObtenerMain()
        except Exception as e:
            print(e)

    def PersonaNaturalObtenerItem(PersonaNaturalId: int):
        try:
            return EntidadDB.PersonaNaturalObtenerItem(PersonaNaturalId)
        except Exception as e:
            print(e)
            
    def PersonaNaturalRegistrar(Ent: PersonaNaturalSaveModel):
        try:
            StartTransaction()
            data = EntidadDB.PersonaNaturalRegistrarDB(Ent)
            EndTransaction()
            return data.PersonaNaturalId
        except Exception as e:
            Restore()
            print(e)
    

    def PersonaNaturalRegistrarEnlace(Ent: DatosClienteItemModel):
        try:
            StartTransaction()
            data = EntidadDB.PersonaNaturalRegistrarEnlaceDB(Ent)
            EndTransaction()
            return data.EntidadId
        except Exception as e:
            Restore()
            print(e)
    
    
    def EmpresaObtenerMain():
        try:
            return EntidadDB.EmpresaObtenerMain()
        except Exception as e:
            print(e)

    def EmpresaObtenerItem(EmpresaId: int):
        try:
            return EntidadDB.EmpresaObtenerItem(EmpresaId)
        except Exception as e:
            print(e)
            
    def EmpresaRegistrar(Ent: EmpresaSaveModel):
        try:
            StartTransaction()
            data = EntidadDB.EmpresaRegistrar(Ent)
            EndTransaction()
            return data.EntidadId
        except Exception as e:
            Restore()
            print(e)

    def EntidadBuscarNombreCompletoItem(Nombre: str):
        try:
            return EntidadDB.EntidadBuscarNombreCompletoItem(Nombre)
        except Exception as e:
            print(e)

    def EntidadObtenerNombreCompletoItem(EntidadId: int):
        try:
            return EntidadDB.EntidadObtenerNombreCompletoItem(EntidadId)
        except Exception as e:
            print(e)

    def EmpresaRegistrarEnlace(Ent: DatosClienteItemModel):
        try:
            StartTransaction()
            data = EntidadDB.EmpresaRegistrarEnlace(Ent)
            EndTransaction()
            return data.EntidadId
        except Exception as e:
            Restore()
            print(e)

File: Facturacion/FactCore/FactCore.BusinessLayer/Entidad.cs
using Framework;
using FactCore.DataLayer;
using FactCore.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactCore.BusinessLayer
{
    public class Entidad
    {

        public static List<EntidadEntity> PersonaNaturalObtenerMain()
        {
            EntidadDB DB = new EntidadDB();
            return DB.PersonaNaturalObtenerMain();
        }
        public static List<EntidadEntity> PersonaNaturalObtenerItem(Int32 PersonaNaturalId)
        {
            EntidadDB DB = new EntidadDB();
            return DB.PersonaNaturalObtenerItem(PersonaNaturalId);
        }
        public static Int32 PersonaNaturalRegistrar(EntidadEntity Ent)
        {
            EntidadDB DB = new EntidadDB();
            DB.PersonaNaturalRegistrar(Ent);

            return Ent.EntidadId;
        }

        public static Int32 PersonaNaturalRegistrarEnlace(EntidadEntity Ent)
        {
            EntidadDB DB = new EntidadDB();
            DB.PersonaNaturalRegistrarEnlace(Ent);

            return Ent.EntidadId;
        }

        public static List<EntidadEntity> EmpresaObtenerMain()
        {
            EntidadDB DB = new EntidadDB();
            return DB.EmpresaObtenerMain();
        }
        public static List<EntidadEntity> EmpresaObtenerItem(Int32 EmpresaId)
        {
            EntidadDB DB = new EntidadDB();
            return DB.EmpresaObtenerItem(EmpresaId);
        }
        public static Int32 EmpresaRegistrar(EntidadEntity Ent)
        {
            EntidadDB DB = new EntidadDB();
            DB.EmpresaRegistrar(Ent);

            return Ent.EntidadId;
        }


        public static List<EntidadEntity> EntidadBuscarNombreCompletoItem(String Nombre)
        {
            EntidadDB DB = new EntidadDB();
            return DB.EntidadBuscarNombreCompletoItem(Nombre);
        }
        public static List<EntidadEntity> EntidadObtenerNombreCompletoItem(Int32 EntidadId)
        {
            EntidadDB DB = new EntidadDB();
            return DB.EntidadObtenerNombreCompletoItem(EntidadId);
        }

        public static Int32 EmpresaRegistrarEnlace(EntidadEntity Ent)
        {
            EntidadDB DB = new EntidadDB();
            DB.EmpresaRegistrarEnlace(Ent);

            return Ent.EntidadId;
        }
    }
}
